### ISSUE TITLE
No sensors detected

### ISSUE THREAD

**Issue by adamgarbo** [sparkfun/OpenLog_Artemis_GNSS_Logger#6]
Hi @PaulZC,

Looking into some strange behaviour with the GNSS Logger firmware on the new OLA hardware revision.

**No sensors detected**
* The beginSensors() function isn't able to detect the GPS-RTK2 board when initially powered up. However, if you enter the menu and immediately exit, you can observe the data to begin being logged. 
* The serial output below illustrates this behaviour.

**No serial output**
* When the GNSS Logger is initially connected via USB, if the Serial Monitor is already open, it will not output any data to the Serial Monitor. It is necessary to close and reopen the Serial Monitor. 
* Is this perhaps due to the changes in deep sleep?

I'm looking deeper into why the I2C device is only detected after the second call to beginSensors() and will keep you posted.

Cheers,
Adam

```
18:14:36.590 -> Config file read complete
18:14:36.696 -> Artemis OpenLog GNSS v1.1
18:14:36.969 -> Finding the next available log file.
18:14:36.969 -> This could take a long time if the SD card contains many existing log files.
18:14:37.071 -> Created log file: dataLog00001.ubx
18:14:37.071 -> SD card online
18:14:37.071 -> Data logging online
18:14:37.071 -> No sensors detected
18:14:46.456 -> 
18:14:46.456 -> Menu: Main Menu
18:14:46.456 -> 1) Configure Logging
18:14:46.456 -> 2) Configure GNSS Device
18:14:46.456 -> 3) Configure Qwiic Bus
18:14:46.456 -> 4) Configure Power Options
18:14:46.456 -> f) Open New Log File
18:14:46.456 -> r) Reset all OLA settings to default
18:14:46.456 -> q) Quit: Close log files and power down
18:14:46.456 -> x) Return to logging
18:14:47.179 -> 
18:14:47.179 -> Returning to logging...
18:14:48.503 -> 2020/08/21 22:14:48.00 UBX Class: 01 ID: 07 NAV-PVT
18:14:48.503 -> 2020/08/21 22:14:48.00 UBX Class: 01 ID: 21 NAV-TIMEUTC
18:14:49.520 -> 2020/08/21 22:14:49.02 UBX Class: 01 ID: 07 NAV-PVT
18:14:49.520 -> 2020/08/21 22:14:49.02 UBX Class: 01 ID: 21 NAV-TIMEUTC
18:14:50.203 -> 2020/08/21 22:14:49.70 UBX Class: 01 ID: 07 NAV-PVT
18:14:50.203 -> 2020/08/21 22:14:49.70 UBX Class: 01 ID: 21 NAV-TIMEUTC
```

**Comment by nseidle** [comment]
Hi Adam - This snuck up on us. The GPS needs a good 1000ms power on before it ack I2C as we expect. OLA v15 firmware got a lot faster, but this is a side affect. I starting poking at the problem in the v16 candidate branch.

**Comment by adamgarbo** [comment]
Hi @nseidle,

I suspected that was the case! With an added delay of 750 ms, the GPS-RTK2 seems to be able to initialize reliably.  

https://github.com/sparkfun/OpenLog_Artemis_GNSS_Logger/blob/72b053b77dd3de98b2a205d0a7739b59e6b405aa/Firmware/OpenLog_Artemis_GNSS_Logging/Sensors.ino#L326

I'm happy to submit a PR to add the delay and make a few other odds and ends changes I have waiting in a branch.

Cheers,
Adam

**Comment by adamgarbo** [comment]
Addressed for the moment by PR #7.
